server/src/main/java/org/candlepin/async/tasks/ImportJob.java

```diff
@@ -0,0 +1,226 @@
+         *
+         * @param uploadedFileName
+         *  the uploaded file name to set for this job
+         *
+         * @return
+         *  a reference to this job config
+         */
+        public ImportJobConfig setUploadedFileName(String uploadedFileName) {
+            this.setJobArgument(UPLOADED_FILE_NAME, uploadedFileName);
+            return this;
+        }
+
+        /**
+         * Sets the conflict overrides the user has specified to this import job.
+         *
+         * @param conflictOverrides
+         *  the conflict overrides to set for this job
+         *
+         * @return
+         *  a reference to this job config
+         */
+        public ImportJobConfig setConflictOverrides(ConflictOverrides conflictOverrides) {
+            this.setJobArgument(CONFLICT_OVERRIDES, conflictOverrides.asStringArray());
+            return this;
+        }
+
+        @Override
+        public void validate() throws JobConfigValidationException {
+            super.validate();
+
+            try {
+                JobArguments arguments = this.getJobArguments();
+
+                String ownerKey = arguments.getAsString(OWNER_KEY);
+                String storedFileId = arguments.getAsString(STORED_FILE_ID);
+                String uploadedFileName = arguments.getAsString(UPLOADED_FILE_NAME);
+
+                if (ownerKey == null || ownerKey.isEmpty()) {
+                    String errmsg = "owner has not been set, or the provided owner lacks a key";
+                    throw new JobConfigValidationException(errmsg);
+                }
+
+                if (storedFileId == null || storedFileId.isEmpty()) {
+                    String errmsg = "stored file id has not been set, or is empty";
+                    throw new JobConfigValidationException(errmsg);
+                }
+
+                if (uploadedFileName == null || uploadedFileName.isEmpty()) {
+                    String errmsg = "uploaded file name has not been set, or is empty";
+                    throw new JobConfigValidationException(errmsg);
+                }
+            }
+            catch (ArgumentConversionException e) {
+                String errmsg = "One or more required arguments are of the wrong type";
+                throw new JobConfigValidationException(errmsg, e);
+            }
+        }
+    }
+
+    private OwnerCurator ownerCurator;
+    private ManifestManager manifestManager;
+
+    @Inject
+    public ImportJob(OwnerCurator ownerCurator, ManifestManager manifestManager) {
+        this.ownerCurator = ownerCurator;
+        this.manifestManager = manifestManager;
+    }
+
+    @Override
+    public Object execute(JobExecutionContext context) throws JobExecutionException {
+        JobArguments args = context.getJobArguments();
+
+        String ownerKey = args.getAsString(OWNER_KEY);
+        ConflictOverrides overrides = new ConflictOverrides(args.getAs(CONFLICT_OVERRIDES, String[].class));
```

Comment:
You should be able to pull the conflict overrides directly without needing to rebuild from the string array. If that's not working, there's a problem in the new serializer paths that we need to address. Ideally the following line should work:
```ConflictOverrides overrides = args.getAs(CONFLICT_OVERRIDES, ConflictOverrides.class)```

If so, the setter above should also change accordingly.

Reply:
This does not work; when setting, the object is not serialized properly.

Reply:
Interesting. We set other objects and pull them out fine. I wonder what's broken with serialization this time...?

In any event, I don't really want to keep adding TODO tasks to the epic, so let me poke around a bit and see how complex it'll be to fix. If it's going to be a large undertaking, we can leave this as-is for now and fix it in another PR. Otherwise if it's a simple fix, I'd rather we do that here instead of working around the problem.

Reply:
Looking into it a bit more, I see why it's serializing the way it is. I've written a lengthy email about this, so what we do here could vary. For now, assume we're going to leave it as written in this PR, and we'll address it as necessary depending on how discussions and planning go.